### ISSUE TITLE
max-len: ignore* rules are too aggressive

### ISSUE THREAD

**Issue by vsemozhetbyt** [eslint/eslint#8560]
* **ESLint Version:** 3.19.0
* **Node Version:** 8.0.0 nightly
* **npm Version:** 4.5.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```js
'use strict';

module.exports = {
  root: true,
  parserOptions: { ecmaVersion: 2017, sourceType: 'script' },
  env: { es6: true, node: true, browser: true },

  rules: {
    'max-len': ['error', {
      code: 80,
      tabWidth: 2,
      ignoreUrls: true,
      ignoreStrings: true,
      ignoreRegExpLiterals: true,
      ignoreTemplateLiterals: true,
    }],
  },
};
```
**What did you do? Please include the actual source code causing the issue.**
```js
'use strict';

console.log('https://short-url.com/', 'short string', /short RegExp/, `short template`);
```
**What did you expect to happen?**
Error message.

**What actually happened? Please include the actual, raw output from ESLint.**
No error message.

This code causes the error message without the `ignore*` rules and does not cause the error message if ANY of the four before mentioned `ignore*` rule is set to `true`. However, the code definitely can be improved in the realm of the rule aim.

Can you add an option like `ignoreIfOnlyOnePerLine` or something, so that the rule is ignored only if the code line can not be wrapped without splitting the only one long entity?

**Comment by danny-andrews** [comment]
Dupe of #8560?

**Comment by vsemozhetbyt** [comment]
@danny-andrews Do you mean https://github.com/eslint/eslint/issues/8189? Yes. maybe. But this one is a bit wider.

Dear collaborators, feel free to close if a dupe.

**Comment by danny-andrews** [comment]
Pshh, yeah, lol.

**Comment by ilyavolodin** [comment]
This is a duplicate of #8189 basically, as soon as ESLint see a match with one of the ignore settings, it ignores the whole line.
